### ISSUE TITLE
Allow duplicate object ID's for pickup/delivery objects in shipments

### ISSUE THREAD

**Issue by dkastl** [VROOM-Project/vroom#539]
Right now it's required to have unique pickup/delivery object ID's in shipments, otherwise an error like this is returned:

```
{"code":2,"error":"Duplicate delivery id: 2."}
```

I have a use case, where many shipments have the same delivery location, and the actual service time is fix for all shipments, that get unloaded at the delivery place from the same vehicle. 

Since the service time is independent from the number of shipments, it can't be specified in the delivery object of each shipment, because the sum of all service times wouldn't be correct.
Also it would make it easier to deal with only one or a few delivery objects, because in the current situation they need to be copied with a different ID into each shipment. 

Is there another way to specify a fix service time for a delivery object, if multiple shipments are brought to the same place?

Examples:
- garbage trucks driving to dump sites
- bus services where passengers board or leave together

The sum of services times in these examples is usually getting too high if defined for each shipment and multiple shipments can be combined.

**Comment by jcoupey** [comment]
We chose to model many-to-one or one-to-many pickup and delivery situations with one `shipment` object per P&D relation so there is no way around duplicating `pickup`/`delivery` objects in input.

I get it it can seem irrelevant to generate different ids for deliveries happening at the same time, but there is not always a guarantee that deliveries at the same place will be grouped. There may be totally valid reasons (think timing or capacity restrictions) to split them in time or even across multiple routes. So we need to avoid any kind of ambiguity here. Also enforcing unique `(type, id)` pairs happened recently when introducing plan mode were your provide the vehicle steps in input, again to avoid any ambiguity. If your `pickup` objects have "natural" unique ids, it's totally valid to use the same id for both `pickup` and `delivery` in a shipment.

>Is there another way to specify a fix service time for a delivery object, if multiple shipments are brought to the same place?

It looks like essentially you're in need of #358. You'd still have to post-process route steps if you want to make multiple tasks at the same place look like a single step, but you'd have a way to get the timing right during optimization.

**Comment by dkastl** [comment]
That's a good point, and thank you for the reference to the issue, which is indeed the same use case. I also thought that it shouldn't be necessary to require all shipments to be delivered at the same time. In fact the extra service time is not really bound to the shipment but to the location. I like the term `setup` for this "location" service time.

This case came up when I started adding shipments to a database table and I realized that I'm adding lots of identical ones that only differ by their ID, and when I apply a service time to the delivery this can lead to very high numbers that might falsely prevent shipments to be grouped together.

One workaround I had in mind was adding this `setup` time to the traveltime to these locations, which would lead to wrong travel times and the service time had to be dededucted again afterwards. However, as long as this extra `setup` cost is the same for all shipments and only bound to the location, it might be possible to implement it that way.

Since the issue is duplicate to #358 I'm closing this one.
